hyracks/hyracks-dataflow-std/src/main/java/org/apache/hyracks/dataflow/std/structures/SerializableVector.java

```
@@ -15,37 +66,76 @@ public SerializableVector(IResetableComparableFactory factory, int recordSize){
      * @param frameSize
      * @param recordSize
      */
-    public SerializableVector(IResetableComparableFactory factory, int frameSize, int recordSize){
-
+    public SerializableVector(int recordSize, int frameSize){
+        frames = new ArrayList<>();
+        init(recordSize);
+        this.frameSize = frameSize;
+        recordPerFrame = frameSize / recordSize;
+        ctx = new FrameManager(frameSize);
     }
 
     @Override
-    public boolean get(int index, IResetableComparable record) {
-        return false;
+    public void get(int index, IResetableSerializable record) {
+        if(index >= sVectorSize) {
+            throw new IndexOutOfBoundsException("index out of bound");
+        }
+        int frameIdx = getFrameIdx(index);
+        int offsetInFrame = getOffsetInFrame(index);
+        record.deserialize(frames.get(frameIdx).array(), offsetInFrame, recordSize);
     }
 
     @Override
-    public boolean append(IResetableComparable record) {
-        return false;
+    public void append(IResetableSerializable record){
+        if(sVectorSize % recordPerFrame == 0){    //add a new frame
+            ByteBuffer frame = null;
+            try {
+                frame = ctx.allocateFrame(frameSize);
+            } catch (HyracksDataException e) {
+                e.printStackTrace();
```

We should throw the exception.